pkg/controller/appdefinition/deploy.go

```diff
@@ -553,6 +555,20 @@ func addPrometheusAnnotations(annotations map[string]string, container v1.Contai
 	}
 }
 
+func addCiliumAnnotations(annotations map[string]string, container v1.Container, cfg *apiv1.Config) {
+	if cfg.UseCiliumLayer7 != nil && *cfg.UseCiliumLayer7 {
```

Comment:
Most of our config stuff is setup so that you don't have to check nil-ness. Is that the case here?

Regardless, can you use `z.Dereference`?